lib/ProgramOptions/Option.h

```diff
@@ -46,6 +46,11 @@ enum class Flags : uint16_t {
                    // target host configuration
   FlushOnFirst = 32,  // when we first see this parameter, we will flush the
                       // contents of its default value before setting it.
+  // also see addOldOption() for renamed and addObsoleteOption() for deprecated
+  // options. for internal, "hidden" options use:
+  //   #ifdef ARANGODB_ENABLE_MAINTAINER_MODE
```

Comment:
Should this rather be `#if defined(ARANGODB_ENABLE_MAINTAINER_MODE)`?
https://github.com/arangodb/arangodb/blob/0cbb9ea895e746892e1c31d5a650b13b68c69632/arangod/Replication2/ReplicatedLog/ReplicatedLogFeature.cpp#L75

Reply:
`#ifdef` and `#if defined` are equivalent, so it can stay as is.
Anyway, the new code comment's position here is a bit misleading, because it is currently located between `FlushOnFist` and `OsLinux`, and both flags don't have anything to do with this comment. Shouldn't it be put somewhere more appropriate?

Reply:
Right, how about placing it after Obsolete or above the enum?

Reply:
near Obsolete seems fine, yes.